modules/apps/commerce/commerce-currency-service/src/main/java/com/liferay/commerce/currency/internal/util/CommercePriceFormatterImpl.java

```diff
@@ -305,6 +330,11 @@ else if (Objects.equals(
 
 	private static final Pattern _commaDecimalPattern = Pattern.compile(
 		"^\\d{1,3}(?:\\.\\d{3})*(?:,\\d+)?$|^\\d+(?:,\\d+)?$");
+	private static final Pattern _negativeCommaDecimalPattern = Pattern.compile(
+		"^-?\\d{1,3}(?:\\.\\d{3})*(?:,\\d+)?$|^\\d+(?:,\\d+)?$");
+	private static final Pattern _negativePeriodDecimalPattern =
+		Pattern.compile(
+			"^-?\\d{1,3}(?:,\\d{3})*(?:\\.\\d+)?$|^\\d+(?:\\.\\d+)?$");
```

Comment:
@brianikim can you give a look at this. The pattern looks to work (it allows negative values), but it is a lot different from your _periodDecimalPattern.